### ISSUE TITLE
Make props accessible without bindings, inspired by React

### ISSUE THREAD

**Issue by dmitriz** [angular/angular.js#13698]
Having seen [how easy it is to access Props in React](https://jsfiddle.net/reactjs/69z2wepo/), I can't help to request the same awesome feature in Angular:

```html
<awesome-comp name="'Trump'"></awesome-comp>
```

I would dream to avoid the clunky binding declarations and have the same level of simplicity:

```js
.component('awesomeComp', {
    template: '<div>You are awesome, Mr. {{$comp.$props.name}}</div>',
})
```

Opinions?

<blockquote class="twitter-tweet" lang="en"><p lang="en" dir="ltr">Make Angular great again.</p>&mdash; Trump on Angular (@realDonaldNG) <a href="https://twitter.com/realDonaldNG/status/655656308545343488">October 18, 2015</a></blockquote>
<script async src="//platform.twitter.com/widgets.js" charset="utf-8"></script>

**Comment by toddmotto** [comment]
Like this idea a lot. One thing I think you're missing here is actually React's `propTypes` inspiration as well:

```js
.component('counter', {
    bindings: {
        count: angular.scopeType.number
    },
    template: '<div>Count: {{counter.props.count}}</div>',
})
```

**Comment by gaui** [comment]
:+1:

**Comment by valorkin** [comment]
Already possible in angular 2 http://victorsavkin.com/post/119943127151/angular-2-template-syntax

**Comment by petebacondarwin** [comment]
This is yet another reason to start the upgrade to Angular 2.

**Comment by toddmotto** [comment]
@petebacondarwin Indeed! Excited for those changes. Some form of support for 1.x apps could potentially be considered though, especially large enterprise apps that will gradually be making code upgrades to get closer and closer to 2 as possible before leaving 1.x!

**Comment by dmitriz** [comment]
@valorkin Does Angular 2 allow the same props-syntax without controller? I could not find it in that article.

**Comment by valorkin** [comment]
@dmitriz no controller - no problem :)
it is available for directives and components

**Comment by dmitriz** [comment]
@valorkin Would you mind giving a more detailed description of what you have in mind? How would you implement the above proposals in Angular 2? I don't see it without controller.

**Comment by petebacondarwin** [comment]
This is similar to #11657
Moving to the ice box for the moment as this is low priority right now.

**Comment by LulzAugusto** [comment]
Hello. I think my question below might have something to do with that feature request, depending on the implementation details.

http://stackoverflow.com/questions/34958681/what-is-the-best-way-to-pass-functions-between-inner-components-in-angularjs-1-5

**Comment by LulzAugusto** [comment]
Please, take into consideration not wrapping functions when implementing that feature. That would be helpful in scenarios like this:
http://stackoverflow.com/questions/34958681/what-is-the-best-way-to-pass-functions-between-inner-components-in-angularjs-1-5

**Comment by petebacondarwin** [comment]
@LulzAugusto  - I don't understand your use case. Can you respond to my comments on SO?

**Comment by LulzAugusto** [comment]
@petebacondarwin I just edited the question. Thanks for your help

**Comment by linhuiw** [comment]
Does angular 4+ version has this feature? @dmitriz @LulzAugusto
